IQSS/dataverse-frontend
# Collection Page Facets Filters

## What this PR does / why we need it:
Adds the facet filtes in the Collection Page.
See [original issue](https://github.com/IQSS/dataverse-frontend/issues/562)

## Which issue(s) this PR closes:

- Closes #562 

## Special notes for your reviewer:
Wait to test until https://github.com/IQSS/dataverse-frontend/pull/563 is merged and I update this branch with develop.

## Suggestions on how to test this:
**Step 1: Run the Development Environment**
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm i && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

**Step 2: Test the feature**
1. Create some collections and datasets in the root collection. Add one or multiple files in one of those created datasets.
2. Navigate to `http://localhost:8000/spa/collections` and in another tab to `http://localhost:8000` and apply same filters in the SPA version and JSF version (search; filter by collection, datasets or files; select and remove facets). Check that they show the same items and facets list.
*Note that for now, for the selected facets button labels, we are only showing the facet value, so when in the JSF version you will see "Author Name: John" in the SPA you will just see "John".

## Does this PR introduce a user interface change? If mockups are available, please link/include them here:
Yes.
<img width="1715" alt="Screen Shot 2024-12-12 at 15 23 01" src="https://github.com/user-attachments/assets/537a874b-9e64-4ade-ae4d-5361f752bbd9" />
<img width="1354" alt="Screen Shot 2024-12-12 at 15 27 42" src="https://github.com/user-attachments/assets/934c9976-c248-4384-876a-73f7f2fdd9f4" />


## Is there a release notes update needed for this change?:
No
## Additional documentation:
No
